babelsberg/constraintinterpreter.js

```diff
@@ -1032,31 +1070,46 @@ Object.subclass('ConstrainedVariable', {
         }
     },
     _searchDefiningSolver: function() {
-            var solver = {weight: -1000, fake: true};
-            this.eachExternalVariableDo(function(eVar) {
-                if (eVar) {
-                    if (!solver.fake) {
-                        this._hasMultipleSolvers = true;
-                    }
-                    var hasEnabledConstraint = false;
-                    for(var i = 0; i < this._constraints.length; i++){
-                        if (this._constraints[i].solver == eVar.solver
-                            && this._constraints[i]._enabled){
-                                hasEnabledConstraint = true;
-                                break;
-                            }
-                    }
+        var solver = {weight: -1000, fake: true, solverName: '(fake)'};
+        var solvers = [];
+        this.eachExternalVariableDo(function(eVar) {
+            if (eVar) {
+                var s = eVar.__solver__;
 
-                    if (this._constraints.length > 0 && !hasEnabledConstraint)
-                        return;
+                if (!solver.fake) {
+                    this._hasMultipleSolvers = true;
+                }
 
-                    var s = eVar.__solver__;
-                    if (s.weight > solver.weight) {
-                        solver = s;
-                    }
+                if (!s.fake) {
+                    solvers.push(s);
                 }
-            }.bind(this));
-            return solver;
+
+                var hasEnabledConstraint = false;
+                for (var i = 0; i < this._constraints.length; i++) {
+                    if (this._constraints[i].solver == s &&
+                        this._constraints[i]._enabled) {
+                            hasEnabledConstraint = true;
+                            break;
+                        }
+                }
+
+                if (this._constraints.length > 0 && !hasEnabledConstraint)
+                    return;
+
+                if (s.weight > solver.weight) {
+                    solver = s;
+                }
+            }
+        }.bind(this));
+
+        if (solver.fake && solvers.length == 1) {
+            // if there is only one disabled solver, use that one even if disabled
+            // why? to make tests green, that's why (because nobody cared about
```

Comment:
before removing this we should properly consider whether this should be in a final version or not. It seems wrong, the rant wants to indicate the tests might rely on incorrect behavior ;)